### PR TITLE
IEP 445: SWTBot Automation

### PR DESCRIPTION
**Pull request by alirana01**
Functional testing update: 
Launchbar CDT selectors and tests around the launch configurations and targets are added

### CODE REVIEW

File: test/.classpath [hunk 1]
```diff
@@ -3,35 +3,22 @@
 	<classpathentry kind="src" path="resources"/>
 	<classpathentry kind="src" path="src"/>
 	<classpathentry kind="src" path="configs"/>
-	<classpathentry exported="true" kind="lib" path="lib/commons-collections4-4.4.jar" sourcepath="lib/commons-collections4-4.4-sources.jar">
-		<attributes>
-			<attribute name="javadoc_location" value="jar:platform:/resource/com.espressif.idf.tests/lib/commons-collections4-4.4-javadoc.jar!/"/>
-		</attributes>
-	</classpathentry>
-	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins">
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-15">
```
Comment (kolipakakondal) [inline test/.classpath:6]: Can we use the base as Java 11?

File: test/.settings/org.eclipse.jdt.core.prefs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=15
+org.eclipse.jdt.core.compiler.compliance=15
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enablePreviewFeatures=disabled
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.problem.reportPreviewFeatures=warning
+org.eclipse.jdt.core.compiler.release=enabled
+org.eclipse.jdt.core.compiler.source=15
```
Comment (kolipakakondal) [inline test/.settings/org.eclipse.jdt.core.prefs:10]: Keep Java compiler compliance level to Java 11

File: test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+	public void creatingNewSerialFlashLaunchTarget()
+	{
+		fixture.givenNewLaunchTargetIsSelected();
+		fixture.givenTargetTypeSelectedIs("Serial Flash Target");
+		fixture.givenNewLaunchTargetName("TestSerialFlash");
+		fixture.givenNewLauncTargetOperatingSystem("LINUX");
+		fixture.givenNewLaunchTargetCPUArchitecture("x86");
+		fixture.givenNewLaunchTargetComPortIsSetTo("COMTEST");
+		fixture.whenFinishIsClicked();
+		fixture.thenLaunchTargetContains("TestSerialFlash");
```
Comment (kolipakakondal) [inline test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java:110]: We don't deal with generic targets

File: test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+	@Test
+	public void creatingNewGDBLaunchTargetOverSerial()
+	{
+		fixture.givenNewLaunchTargetIsSelected();
+		fixture.givenTargetTypeSelectedIs("GDB Remote Serial");
+		fixture.givenNewLaunchTargeGDBName("TestGDBSerial");
+		fixture.whenFinishIsClicked();
+		fixture.thenLaunchTargetContains("TestGDBSerial");
+		fixture.thenLaunchTargetIsSelectedFromLaunchTargets("TestGDBSerial");
+	}
```
Comment (kolipakakondal) [inline test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java:136]: We need only esp launch target, I think we can remove this as well

File: test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+	{
+		fixture.givenNewLaunchTargetIsSelected();
+		fixture.givenTargetTypeSelectedIs("GDB Remote TCP");
+		fixture.givenNewLaunchTargeGDBTcpName("TestGdbTcp");
+		fixture.givenHostNameOrIpIs("127.0.0.1");
+		fixture.givenPortNumberIs(8087);
+		fixture.whenFinishIsClicked();
+		fixture.thenLaunchTargetContains("TestGdbTcp");
+		fixture.thenLaunchTargetIsSelectedFromLaunchTargets("TestGdbTcp");
+	}
```
Comment (kolipakakondal) [inline test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java:149]: We need only esp launch target, I think we can remove this as well

**Review by kolipakakondal — APPROVED** [review]
LGTM

We can remove log4j jar, as we are not using it anywhere.

File: test/.classpath (hunk 1 above)
Comment (alirana01) [reply test/.classpath]: Done

File: test/.settings/org.eclipse.jdt.core.prefs (hunk 2 above)
Comment (alirana01) [reply test/.settings/org.eclipse.jdt.core.prefs]: Done

File: test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java (hunk 3 above)
Comment (alirana01) [reply test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java]: removed

File: test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java (hunk 4 above)
Comment (alirana01) [reply test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java]: removed

File: test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java (hunk 5 above)
Comment (alirana01) [reply test/src/com/espressif/idf/test/executable/cases/launchconfiguration/LaunchBarCDTConfigurationsTest.java]: removed

**Comment by alirana01** [comment]
@kolipakakondal the review comments were addressed please proceed with merge and close this branch after as well.

**Review by kolipakakondal — APPROVED** [review]

**Pull request merged**